pkg/lib/accountmanagement/service.go

```diff
@@ -290,3 +320,90 @@ func (s *Service) ChangePassword(input *ChangePasswordInput) (*ChangePasswordOut
 	return &ChangePasswordOutput{RedirectURI: redirectURI}, nil
 
 }
+
+func (s *Service) AddPasskey(input *AddPasskeyInput) (*AddPasskeyOutput, error) {
+	userID := input.Session.GetAuthenticationInfo().UserID
+	creationResponse := input.CreationResponse
+	creationResponseBytes, err := json.Marshal(creationResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	authenticatorSpec := &authenticator.Spec{
+		UserID: userID,
+		Kind:   authenticator.KindPrimary,
+		Type:   model.AuthenticatorTypePasskey,
+		Passkey: &authenticator.PasskeySpec{
+			AttestationResponse: creationResponseBytes,
+		},
+	}
+	authenticatorID := uuid.New()
+	authenticatorInfo, err := s.Authenticators.NewWithAuthenticatorID(authenticatorID, authenticatorSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	identitySpec := &identity.Spec{
+		Type: model.IdentityTypePasskey,
+		Passkey: &identity.PasskeySpec{
+			AttestationResponse: creationResponseBytes,
+		},
+	}
+	identityInfo, err := s.Identities.New(userID, identitySpec, identity.NewIdentityOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.Database.WithTx(func() error {
+		err = s.Identities.Create(identityInfo)
+		if err != nil {
+			return err
+		}
+		err = s.Authenticators.Create(authenticatorInfo, false)
+		if err != nil {
+			return err
+		}
+		err = s.PasskeyService.ConsumeAttestationResponse(creationResponseBytes)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &AddPasskeyOutput{}, nil
+}
+
+func (s *Service) RemovePasskey(input *RemovePasskeyInput) (*RemovePasskeyOutput, error) {
+	identityID := input.IdentityID
+	var identityInfo *identity.Info
+
+	err := s.Database.WithTx(func() (err error) {
+		identityInfo, err = s.Identities.Get(identityID)
+		if err != nil {
+			return
+		}
+
+		if identityInfo.UserID != input.Session.GetAuthenticationInfo().UserID {
+			return api.NewInvariantViolated(
+				"IdentityNotBelongToUser",
+				"identity does not belong to the user",
+				nil,
+			)
+		}
+
+		err = s.Identities.Delete(identityInfo)
+		if err != nil {
+			return
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &RemovePasskeyOutput{}, nil
+}
```

Comment:
Please list out all the possible flows of nodes and edges on `interaction` and check carefully what side effects we need to perform when removing a new identity